libsolidity/formal/CVC4Interface.cpp

```diff
@@ -132,8 +132,17 @@ CVC4::Expr CVC4Interface::toCVC4Expr(Expression const& _expr)
 		else if (n == "false")
 			return m_context.mkConst(false);
 		else
-			// We assume it is an integer...
+		{
+			try
+			{
+				bigint isNumber(n);
+			}
+			catch (...)
```

Comment:
Any reason for not just letting the exception be thrown directly? ``solAssert(false, "");`` will only throw another one anyways... not sure about it. Generally I'm still not sure whether this is the nicest way of checking for numbers... And: doesn't ``CVC4::Rational(n)`` throw already if it fails to parse? Looks like it from peeking at the source and looking at the underlying GMP docs...